### PR TITLE
[rb] Upload file to browser when file detector set in w3c bridge

### LINKED ISSUE #6064: [rb] send_keys_to_element method in W3C bridge does not support file detector
## Meta -
OS:  
Any

Selenium Version:  
3.12.0

Browser:  
Firefox (Remote Driver) which using w3c protocol

Browser Version:  
49+

## Expected Behavior -

Using file_detector and send_keys method I can upload file  from local machine to remote browser.

## Actual Behavior -

Ruby binding for w3c protocol does not support file_detector. File not upload. Client get error "File not found: <path\to\file>"

OSS bridge with file_detector support: https://github.com/SeleniumHQ/selenium/blob/master/rb/lib/selenium/webdriver/remote/oss/bridge.rb#L365

W3C without file_detector support: https://github.com/SeleniumHQ/selenium/blob/master/rb/lib/selenium/webdriver/remote/w3c/bridge.rb#L362

### PR DESCRIPTION
**Pull request by mskvn**
Fixes #6064

- [X] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

### CODE REVIEW

File: rb/lib/selenium/webdriver/remote/w3c/bridge.rb [hunk 1]
```diff
@@ -358,13 +358,25 @@ def click_element(element)
...
+              keys = [upload(local_file)] if local_file
+            end
             # Keep .split(//) for backward compatibility for now
             text = keys.join('')
             execute :element_send_keys, {id: element}, {value: text.split(//), text: text}
           end
 
+          def upload(local_file)
+            unless File.file?(local_file)
+              raise Error::WebDriverError, "you may only upload files: #{local_file.inspect}"
```
Comment (luke-hill) [inline rb/lib/selenium/webdriver/remote/w3c/bridge.rb]: I'd alter this message to be a bit more verbose. Here you're taking an input which you know will only be called when file_detector is set. But your message says you may only upload files (Which is what you're doing)

Comment (mskvn) [reply rb/lib/selenium/webdriver/remote/w3c/bridge.rb]: Sorry, but I not understand what you mean :) This check only show that local_file is file or not. How message should be verbose? 

I have only one idea. Add to message something like this: "File detector set, but return not file" or "File detector should return file". But in this case maybe this check should move after file_detector.call

P.S: This part of code I copied from oss bridge :)

Comment (luke-hill) [reply rb/lib/selenium/webdriver/remote/w3c/bridge.rb]: How about...

"You are trying to work with something that isn't a file." (Don't really need object inspect here)
"File detector only works with Files. #{local_file.inspect} isnt a file!" <- Obj inspect makes sense here.

P.s. if this is copied from OSS Bridge I'd probably fix it up there too.

Comment (mskvn) [reply rb/lib/selenium/webdriver/remote/w3c/bridge.rb]: Got it. Fixed :)

**Comment by luke-hill** [comment]
To OP @mskvn one failure in one of the travis runs is related to this potentially. Might need to fix up some specs related to your change or have a look

Start here: https://travis-ci.org/SeleniumHQ/selenium/jobs/399531688

**Comment by mskvn** [comment]
Yep, sorry, made typo in constant name :(
Already fixed.

**Pull request closed without merging**